src/Evolution/Systems/NewtonianEuler/Limiters/KxrcfTci.hpp

```diff
@@ -0,0 +1,201 @@
+    const size_t index_of_edge =
+        (dir.side() == Side::Lower ? 0 : mesh.extents()[sliced_dim] - 1);
+    const auto momentum_on_edge = data_on_slice(
+        cons_momentum_density, mesh.extents(), sliced_dim, index_of_edge);
+    const auto momentum_dot_normal =
+        dot_product(momentum_on_edge, unnormalized_normals.at(dir));
+
+    // Skip boundaries with no significant inflow
+    // Note: the cutoff value here is small but arbitrarily chosen.
+    if (min(get(momentum_dot_normal)) > -1e-12) {
+      continue;
+    }
+    inflow_boundaries_present = true;
+
+    // This mask has value 1. for momentum_dot_normal < 0.
+    //                     0. for momentum_dot_normal >= 0.
+    const DataVector inflow_mask = 1. - step_function(get(momentum_dot_normal));
+    // Mask is then weighted pointwise by the Jacobian determinant giving
+    // surface integrals in inertial coordinates. This Jacobian determinant is
+    // given by the product of the volume Jacobian determinant with the norm of
+    // the unnormalized face normals.
+    const DataVector weighted_inflow_mask =
+        inflow_mask *
+        get(data_on_slice(det_logical_to_inertial_jacobian, mesh.extents(),
+                          sliced_dim, index_of_edge)) *
+        get(magnitude(unnormalized_normals.at(dir)));
+
+    inflow_area +=
+        definite_integral(weighted_inflow_mask, mesh.slice_away(sliced_dim));
+
+    // This is the step that is incompatible with h/p refinement. For use with
+    // h/p refinement, would need to correctly obtain the neighbor solution on
+    // the local grid points.
+    const auto neighbor_vars_on_edge =
+        data_on_slice(neighbor_and_data.second.volume_data, mesh.extents(),
+                      sliced_dim, index_of_edge);
+
+    const auto density_on_edge = data_on_slice(
+        cons_mass_density, mesh.extents(), sliced_dim, index_of_edge);
+    const auto& neighbor_density_on_edge =
+        get<NewtonianEuler::Tags::MassDensityCons>(neighbor_vars_on_edge);
+    inflow_delta_density += definite_integral(
+        (get(density_on_edge) - get(neighbor_density_on_edge)) *
+            weighted_inflow_mask,
+        mesh.slice_away(sliced_dim));
+
+    const auto energy_on_edge = data_on_slice(
+        cons_energy_density, mesh.extents(), sliced_dim, index_of_edge);
+    const auto& neighbor_energy_on_edge =
+        get<NewtonianEuler::Tags::EnergyDensity>(neighbor_vars_on_edge);
+    inflow_delta_energy +=
+        definite_integral((get(energy_on_edge) - get(neighbor_energy_on_edge)) *
+                              weighted_inflow_mask,
+                          mesh.slice_away(sliced_dim));
+  }
+
+  if (not inflow_boundaries_present) {
+    // No boundaries had inflow, so not a troubled cell
+    return false;
+  }
+
+  // KXRCF take h to be the radius of the circumscribed circle
+  const double h = 0.5 * magnitude(element_size);
+  const double h_pow = pow(h, 0.5 * mesh.extents(0));
```

Comment:
please explain what this is; why use the extents of the lowest dim and not the min/mean/max of the dims?

Reply:
I'll add a check that the extents are uniform, because the reference doesn't specify how to handle non-uniform extents